### PR TITLE
Video metric

### PR DESCRIPTION
**Pull request by Songyuanwei**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

Add metrics to measure the quality of generated videos.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: mindone/tools/eval/eval_clip_score_video.py
Comment (hadipash) [inline mindone/tools/eval/eval_clip_score_video.py:1]: Can this functionality be implemented as a function as well so that one can call it within a python script?

**Review by zhtmike — APPROVED** [review]

File: mindone/tools/eval/eval_clip_score_video.py
Comment (Songyuanwei) [reply mindone/tools/eval/eval_clip_score_video.py:1]: done

Comment (hadipash) [reply mindone/tools/eval/eval_clip_score_video.py:1]: I think would be better to implement it as a class that can be easily instantiated and used to iterate over videos using `clip_score_text` or `clip_score_frame`.

Comment (Songyuanwei) [reply mindone/tools/eval/eval_clip_score_video.py:1]: done

File: mindone/tools/eval/eval_clip_score_video.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# eval clip score for video
+import argparse
```
Comment (CaitinZhao) [inline mindone/tools/eval/eval_clip_score_video.py:2]: 放metrics下面

File: mindone/transformers/models/clip/modeling_ms_clip.py [hunk 2]
```diff
@@ -94,7 +95,7 @@ def _make_causal_mask(
         Make causal mask used for bi-directional self-attention.
         """
         bsz, tgt_len = input_ids_shape
-        mask = ops.full((tgt_len, tgt_len), float("-inf"))
+        mask = ops.full((tgt_len, tgt_len), np.finfo(np.float32).min, dtype=dtype)
```
Comment (geniuspatrick) [inline mindone/transformers/models/clip/modeling_ms_clip.py:98]: 这个为啥要改？

File: mindone/transformers/models/clip/modeling_ms_clip.py [hunk 3]
```diff
@@ -804,3 +805,136 @@ def construct(
 
         outputs = (image_embeds, vision_outputs[0]) + vision_outputs[2:]
         return tuple(output for output in outputs if output is not None)
+
+
+class CLIPModel(CLIPPreTrainedModel):
```
Comment (geniuspatrick) [inline mindone/transformers/models/clip/modeling_ms_clip.py:810]: 放到和huggingface一样的位置

File: mindone/transformers/models/clip/modeling_ms_clip.py [hunk 4]
```diff
@@ -804,3 +805,136 @@ def construct(
...
+    def get_text_features(
+        self,
+        input_ids: Optional[ms.Tensor] = None,
+        attention_mask: Optional[ms.Tensor] = None,
+        position_ids: Optional[ms.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+    ) -> ms.Tensor:
+        # Use CLIP model's config for some fields (if specified) instead of those of vision & text components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
```
Comment (geniuspatrick) [inline mindone/transformers/models/clip/modeling_ms_clip.py:855]: `self.config.output_attentions`直接使用会有问题，参见已有代码的写法

File: mindone/transformers/models/clip/modeling_ms_clip.py [hunk 5]
```diff
@@ -804,3 +805,136 @@ def construct(
...
+        self,
+        input_ids: Optional[ms.Tensor] = None,
+        pixel_values: Optional[ms.Tensor] = None,
+        attention_mask: Optional[ms.Tensor] = None,
+        position_ids: Optional[ms.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+    ) -> Tuple:
+        # Use CLIP model's config for some fields (if specified) instead of those of vision & text components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
```
Comment (geniuspatrick) [inline mindone/transformers/models/clip/modeling_ms_clip.py:906]: `self.config`

File: mindone/transformers/models/clip/modeling_ms_clip.py (hunk 2 above)
Comment (Songyuanwei) [reply mindone/transformers/models/clip/modeling_ms_clip.py:98]: 如果不改的话，mask的值是-9223372036854775808，
图片1
会和torch生成的结果不一致，最后texttransformer生成的结果误差会大一些。改掉之后，mask的值是-3.4028235e+38
图片2，与torch一致。。

File: mindone/transformers/models/clip/modeling_ms_clip.py (hunk 3 above)
Comment (Songyuanwei) [reply mindone/transformers/models/clip/modeling_ms_clip.py]: done

File: mindone/transformers/models/clip/modeling_ms_clip.py (hunk 4 above)
Comment (Songyuanwei) [reply mindone/transformers/models/clip/modeling_ms_clip.py]: done

File: mindone/transformers/models/clip/modeling_ms_clip.py (hunk 5 above)
Comment (Songyuanwei) [reply mindone/transformers/models/clip/modeling_ms_clip.py]: done

**Review by geniuspatrick — APPROVED** [review]
